src/platform/silabs/multi-ota/SiWx917/OTAWiFiFirmwareProcessor.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <platform/silabs/multi-ota/OTAMultiImageProcessorImpl.h>
+#include <platform/silabs/multi-ota/SiWx917/OTAWiFiFirmwareProcessor.h>
+
+#include <app/clusters/ota-requestor/OTARequestorInterface.h>
+
+extern "C" {
+#include "sl_si91x_driver.h"
+}
+
+namespace chip {
+
+CHIP_ERROR OTAWiFiFirmwareProcessor::ProcessInternal(ByteSpan & block)
+{
+    sl_status_t status = SL_STATUS_OK;
+
+    ChipLogProgress(SoftwareUpdate, "Process WiFi Block");
+    // The first block is expected to be the OTA descriptor.
+    // `ProcessDescriptor` handles it and sets `mDescriptorProcessed` to avoid reprocessing.
+    if (!mDescriptorProcessed)
+    {
+        ReturnErrorOnFailure(ProcessDescriptor(block));
+#if OTA_ENCRYPTION_ENABLE
+        /* 16 bytes to used to store undecrypted data because of unalignment */
+        mAccumulator.Init(requestedOtaMaxBlockSize + 16);
+#endif // OTA_ENCRYPTION_ENABLE
+    }
+
+#if OTA_ENCRYPTION_ENABLE
+    MutableByteSpan byteblock = MutableByteSpan(mAccumulator.data(), mAccumulator.GetThreshold());
+    memcpy(&byteblock[0], &byteblock[requestedOtaMaxBlockSize], mUnalignmentNum);
+    memcpy(&byteblock[mUnalignmentNum], block.data(), block.size());
+
+    if (mUnalignmentNum + block.size() < requestedOtaMaxBlockSize)
+    {
+        uint32_t alignmentnum = (mUnalignmentNum + block.size()) / 16;
+        alignmentnum          = alignmentnum * 16;
+        mUnalignmentNum       = (mUnalignmentNum + block.size()) % 16;
+        memcpy(&byteblock[requestedOtaMaxBlockSize], &byteblock[alignmentnum], mUnalignmentNum);
+        byteblock.reduce_size(alignmentnum);
+    }
+    else
+    {
+        mUnalignmentNum = mUnalignmentNum + block.size() - requestedOtaMaxBlockSize;
+        byteblock.reduce_size(requestedOtaMaxBlockSize);
+    }
+
+    OTATlvProcessor::vOtaProcessInternalEncryption(byteblock);
+    block = byteblock;
```

Comment:
This is duplicated code. My same comments apply here and this should probably be made into a common function  to both processors